### PR TITLE
PR #4 : Implement Get Item by ID (Admin and User) 

### LINKED ISSUE #4: Issue #4 : Implement Get Item by ID (Admin and User)
**Description:**  
- **Task:** Implement the function to retrieve a specific to-do list item by its `ID` (which should be a `UUID`). This operation should be accessible to users with either the `ROLE_ADMIN` or `ROLE_USER` roles.
- **Endpoint:** `GET /api/todo/{id}`
- **Path Parameter:** `id` (UUID)
- **Response Example:**
  ```json
  {
    "id": "c1b8f5b0-7d91-4b23-abc5-8eeb9cd4c721",
    "name": "Sample Item"
  }
  ```

---

**Access Control:**  
- **Only users with `ROLE_ADMIN` or `ROLE_USER`** can retrieve a to-do item by its ID.
- Implement security checks in the controller/service layer to enforce this role restriction.
- Ensure that unauthorized users attempting this action receive a `403 Forbidden` response.

- **Error Response Example (Unauthorized):**
  ```json
  {
    "error": "Forbidden",
    "message": "You do not have permission to access this item."
  }
  ```

---

**Test:**  
- Implement **JUnit** and **Integration Tests** for the following scenarios:
  1. **Successful Retrieval by ID:**
     - Ensure that users with `ROLE_ADMIN` or `ROLE_USER` can successfully retrieve a to-do item by its UUID.
     - Verify that the correct item is returned in the response body.
  2. **Unauthorized Access:**
     - Ensure that users without the required roles (or unauthenticated users) receive a `403 Forbidden` error when attempting to access an item by ID.
  3. **Item Not Found:**
     - Test the behavior when an invalid or non-existent UUID is provided, ensuring that the appropriate `404 Not Found` response is returned.
     
- **Error Response Example (Not Found):**
  ```json
  {
    "error": "Not Found",
    "message": "The to-do item with the specified ID does not exist."
  }
  ```

---

### PR DESCRIPTION
**Pull request by artfulCoder98**
### Description

Please provide a brief description of the changes made in this pull request and the issue(s) it addresses.

Implement the function to retrieve a specific to-do list item by its ID (which should be a UUID). This operation should be accessible to users with either the `ROLE_ADMIN` or `ROLE_USER` roles.

#### Related Issue

Link to the issue this pull request addresses: #[issue number]

### Checklist

Before submitting your pull request, ensure the following:

- [x] **Title and Branch Naming Conventions:**

    - The pull request title follows the
      standard: [Pull Request Naming Conventions](CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](CONTRIBUTING.md#branch-naming-conventions).

- [x] **Local Testing:**

    - I have tested my changes locally on Postman, and they are working as expected.

- [x] **Code Quality:**

    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [x] **Documentation:**

    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes
      made in Postman.

- [x] **Testing:**

    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [x] **Reviewers and Assignees:**

    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [x] **Labels and Associations:**
    - The pull request is associated with its related issue.
    - Appropriate labels have been applied.

### CODE REVIEW

File: src/main/java/com/example/todowithcouchbase/common/exception/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -67,10 +69,12 @@ protected ResponseEntity<Object> handlePathVariableErrors(final ConstraintViolat
                 );
 
         CustomError customError = CustomError.builder()
+                .time(LocalDateTime.now())
                 .httpStatus(HttpStatus.BAD_REQUEST)
                 .header(CustomError.Header.VALIDATION_ERROR.getName())
                 .message("Constraint violation")
                 .subErrors(subErrors)
+                .isSuccess(false)
```
Comment (moaydogdu) [inline src/main/java/com/example/todowithcouchbase/common/exception/GlobalExceptionHandler.java]: Bu builder'da tekrarlayan şekilde time ve isSuccess alanları aynı değerler ile setlenmiş. Kod tekrarının önüne geçmek için bu iki alana `@Builder.Default` ile default değerler atayabiliriz.

**Review by moaydogdu — COMMENTED** [review]
Referansı değişmediği halde final kullanılmayan çok fazla nesne var.

File: src/main/java/com/example/todowithcouchbase/common/exception/GlobalExceptionHandler.java (hunk 1 above)
Comment (Rapter1990) [reply src/main/java/com/example/todowithcouchbase/common/exception/GlobalExceptionHandler.java]: Gerekli düzeltmeler yapıldı.

Comment (artfulCoder98) [reply src/main/java/com/example/todowithcouchbase/common/exception/GlobalExceptionHandler.java]: Kontrolleri sağlayıp final değişkenler için proje bazında düzenlemeler yapılacak tamamdır

**Review by suhacanuluer — APPROVED** [review]

**Review by Rapter1990 — APPROVED** [review]

**Pull request merged**